### PR TITLE
Install some gem dependencies and fix rspec

### PR DESCRIPTION
**Pull request by jonathanchiu**
This PR addresses installing the following gems
* state_machines
* factory_girl_rails

This also addresses fixing RSpec, namely:
* Including the rspec gem in both development and test environments
* Initializing the application with the default RSpec configuration files so that specs will execute properly
* Formatting RSpec output in terminal so that it is more descriptive and meaningful, example output below

screen shot 2016-03-03 at 4 52 10 pm

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -62,10 +62,6 @@
   #   - http://rspec.info/blog/2014/05/notable-changes-in-rspec-3/#zero-monkey-patching-mode
   config.disable_monkey_patching!
 
-  # This setting enables warnings. It's recommended, but in some cases may
-  # be too noisy due to issues in dependencies.
-  config.warnings = true
```
Comment (caramcc) [inline spec/spec_helper.rb]: I take it warnings were too noisy?

Comment (jonathanchiu) [inline spec/spec_helper.rb]: This was actually the default generated `spec_helper.rb` from RSpec via `rails generate rspec:install` :8ball:

**Comment by ThePletch** [comment]
LGTM :+1: 

We should look into setting up the spec runner so it clears the database between specs, though. That always bites me on Rails projects.

File: spec/spec_helper.rb (hunk 1 above)
Comment (caramcc) [inline spec/spec_helper.rb]: Interesting. Works for me ¯\\\_(ツ)_/¯

**Pull request merged**